### DISCUSSION TITLE
How does ShapeCrawler handle SmartArt?

### DISCUSSION THREAD

**Discussion by pucx** [ShapeCrawler/ShapeCrawler#1173] [Q&A]
In the ShapeCrawler library, are there functions or classes available to handle SmartArt, or are there plans to add support functions?

**Comment by AdamShakhabov** [comment]
No, the library doesn't support SmartArt yet, but yes, it is planned.

**Comment by AdamShakhabov** [comment]
Here is the current SmartArt Support in ShapeCrawler:

1. **Creating new SmartArt shapes** — Only `SmartArtType.BasicBlockList`:
   ```csharp
   var smartArtShape = slide.Shapes.AddSmartArt(x, y, width, height, SmartArtType.BasicBlockList);
   ```

2. **Detecting SmartArt in existing presentations** — ShapeCrawler can identify SmartArt shapes when loading presentations.

3. **Node management for newly created SmartArt:**
   - Add nodes: `smartArt.Nodes.AddNode("Text")`
   - Modify node text: `node.Text = "New Text"`
   - Enumerate nodes: `foreach (var node in smartArt.Nodes)`
   - Get node count: `smartArt.Nodes.Count`

**Example usage (for newly created SmartArt only):**
```csharp
var pres = new Presentation(p => p.Slide());
var smartArt = pres.Slide(1).Shapes.AddSmartArt(50, 50, 400, 300, SmartArtType.BasicBlockList).SmartArt!;

// Add nodes
var node1 = smartArt.Nodes.AddNode("First Item");
var node2 = smartArt.Nodes.AddNode("Second Item");

// Modify text
node1.Text = "Updated First Item";

pres.Save("output.pptx");
```
